packages/validation/schema.type.test.js

```diff
@@ -28,8 +28,43 @@ const { testCases, inMemoryFiles } = createTestCases({
       const result = schema.parse({ user: { name: 'test' } })
       expectAssignable<{ user: { name: string } }>(result)
     `,
+    'number() returns a number': `
+      const validator = number()
+      const result = validator.parse(123)
+      expectAssignable<number>(result)
+    `,
+    'array(string()) returns a string[]': `
+      const validator = array(string())
+      const result = validator.parse(['a', 'b', 'c'])
+      expectAssignable<string[]>(result)
```

Comment:
I rather use the `Array` notation:

```suggestion
      expectAssignable<Array<string>>(result)
```

This same problem exists in other places as well